File: daily-coding-problems/5/main.ts
function cons(a: number, b: number): Function
{
    const pair = (f: Function): number =>
    {
        return f(a, b);
    };

    return pair;
}


/* eslint-disable @typescript-eslint/no-unused-vars */
function car(f: Function): Function
{
    // @ts-ignore
    return f((a: number, b: number): number => { return a; });
}


function cdr(f: Function): Function
{
    // @ts-ignore
    return f((a: number, b: number): number => { return b; });
}
/* eslint-enable */


/* eslint-disable no-console */
console.log(car(cons(3, 4)));
console.log(cdr(cons(3, 4)));
/* eslint-enable */

File: daily-coding-problems/5/main.py
#!/usr/bin/env python3


def cons(a, b):
    def pair(f):
        return f(a, b)

    return pair


def car(f):
    x = lambda a, b: a
    return f(x)


def cdr(f):
    x = lambda a, b: b
    return f(x)


print(car(cons(3, 4)))
print(cdr(cons(3, 4)))
